### PR TITLE
Update 02-04-01-Layouts.md

### PR DESCRIPTION
**Pull request by kezek**
adding some stuff about Layout

### CODE REVIEW

File: _posts/02-04-01-Layouts.md [hunk 1]
```diff
@@ -6,4 +6,6 @@ isChild: true
 
 * Use `<update handle="" />` to refactor repeated block declarations.
 * Move blocks from one reference to another using `<action method="unsetChild"></action>` and `<action method="insert"></action>`.
-* Create a `local.xml` in your theme as a place to store all of your theme modifications.  Don't create `catalog.xml`, `customer.xml`, `page.xml`, etc, files unless absolutely necessary.
\ No newline at end of file
+* ~~Create a `local.xml` in your theme as a place to store all of your theme modifications.  Don't create `catalog.xml`, `customer.xml`, `page.xml`, etc, files unless absolutely necessary.~~
+* Avoid adding modifications to a `local.xml`: XML nodes under `local.xml` are final (cannot be overridden) , since they are merged last by the Config Model ; there are no 'generic' changes in Magento , since each functionality should have it's own Module
```
Comment (punkstar) [inline _posts/02-04-01-Layouts.md:10]: Hi - Could you clarify this line a bit more?  For example, "Avoid adding modifications to a `local.xml`?  Modifications of what?

Comment (kezek) [inline _posts/02-04-01-Layouts.md:10]: Sorry , was in a bit of a hurry and couldn't find the right words . Any layout related changes shouldn't be injected in any 'default' magento layout files , like you specified previously (i.e. catalog.xml , customer.xml and local.xml being one of them as well).
1. Because theme fallback mechanism will ensure that the layout files from the current theme will be merged , along with those that have been injected via layout update , but are not found under the current theme 'layout' folder.
2. Nodes under 'local.xml' will be merged at the end of the process (for which the Config Model is responsible) , meaning that they are 'final' (similar to final keyword in PHP).

For illustration :
theme A (default)
theme B (mytheme) - both under the same package , theme B being the current theme used

Both themes have a local.xml defined , but the nodes from theme A are not merged (which could be a good thing , or not) . Worse case is when you have a module with a layout update which needs specific layout changes that are found under local.xml . This means committing a revision under this module scope that changes local.xml , which in my view is not ideal (all layout related logic should be encapsulated under the module's layout file which has been injected via layout update in config.xml).

Let me know what you think.

Comment (Flyingmana) [inline _posts/02-04-01-Layouts.md:10]: I think the problem here is mixing rules for module development and rules for theme development.

A module should never create/touch the layout/local.xml

But if you create a theme for your webshop, then you use only the layout/local.xml

Also note, creating a generic reusable theme counts as module

**Comment by punkstar** [comment]
This prose was a little hard to follow, so I tried to condense the discussion about into a new pull request: #11.  Could you guys check it out and let me know what you think?
